recipes/linux-headers-generic/all/conanfile.py

```diff
@@ -0,0 +1,38 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class LinuxHeadersGenericConan(ConanFile):
+    name = "linux-headers-generic"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.kernel.org/"
+    license = "GPL-2.0-only"
+    description = "Generic Linux kernel headers"
+    topics = ("linux", "headers", "generic")
+    settings = "os", "arch"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("linux-headers-generic supports only Linux")
+        if hasattr(self, "settings_build") and tools.cross_building(self):
+            raise ConanInvalidConfiguration("linux-headers-generic can not be cross-compiled")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def build(self):
+        with tools.chdir(os.path.join(self._source_subfolder)):
+            autotools = AutoToolsBuildEnvironment(self)
+            autotools.make(target="headers")
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        self.copy("include/*.h", src=os.path.join(self._source_subfolder, "usr"))
```

Comment:
okay, I hope all the headers have ".h" extension in Linux.

Reply:
I checked 😀 the include dir also contains makefiles and some other stuff which I guess we don't want

Reply:
Not sure if you could use DESTDIR and avoid copying all header, I mean, installing directly to the package folder.

Reply:
I'll check if that's possible

Reply:
@sparik Yes. make headers only install headers, so it's safe.

Reply:
only headers. but you still can remove extra files. we usually run install and remove extra file.

Reply:
The situation is like this: `make headers_install` accepts a `INSTALL_HDR_PATH` argument which could be set to `self.package_folder`. In that case, `INSTALL_HDR_PATH` would only contain header files. However, we cannot run `make headers_install` because it requires `rsync`, so we are left with running `make headers` and manually copying `*.h` files to the package folder, which is what the recipe does now. So I think we should leave it like this.

Reply:
yes, sounds good for me